src/poliastro/twobody/propagation.py

```diff
@@ -100,17 +86,29 @@ def cowell(orbit, tof, rtol=1e-10, *, ad=None, jac=None, **ad_kwargs):
 
     # Set the non Keplerian acceleration
     if ad is None:
-        jac = functools.partial(gravitational_jac, k=k)
         ad = lambda t0, u_, k_: (0, 0, 0)
 
     f_with_ad = functools.partial(func_twobody, k=k, ad=ad, ad_kwargs=ad_kwargs)
-    rr = solve_ivp(f_with_ad, (0, tof), u0, t_eval=[tof], rtol=rtol, atol=1e-12, method=DOP835)
-    if rr.success:
-        r, v = rr.y[:3], rr.y[3:]
-    else:
+
+    if not hasattr(tof, "__len__"):
+        tof = [tof]
+
+    result = solve_ivp(f_with_ad, (0, max(tof)), u0,
+                       rtol=rtol, atol=1e-12, method=DOP835,
+                       dense_output=True)
+    if not result.success:
         raise RuntimeError("Integration failed")
 
-    return r, v
+    rrs = []
+    vvs = []
+    for i in range(len(tof)):
+        y = result.sol(tof[i])
+        rrs.append(y[:3])
+        vvs.append(y[3:])
+
+    if len(tof) == 1:
```

Comment:
If the user propagates with a list of times, I would always expect a list, even if the input list has only one element. So I would not make a special case here.

Reply:
But we have to! The problem is, there are functions that propagate `cowell` not with the list, but with single time value (for instance come tests). And the output is expected to be, once again, single orbit. So, we need to track whether user passed list or single value to cowell and act accordingly.

What I wrote above is not exactly this, I admit: if user gives list, we should return list. But we have to, anyway, work with the special case when user passed just one single value not in the list.

Reply:
I know, and that's why we have to distinguish between scalar value and list. But still you added a third case, which is converting a list of one element to the scalar case. As you said, if the user gives a list we should return a list.